### DISCUSSION TITLE
Async Insert batch

### DISCUSSION THREAD

**Discussion by 0x366** [ClickHouse/clickhouse-go#1355] [Q&A]
Trying to async insert batch. Found an example of async insert with one row
```go
	if err := d.conn.AsyncInsert(ctx, `INSERT INTO example VALUES (
		?, ?, ?, now()
	)`, false, i, "Golang SQL database driver", []uint8{1, 2, 3, 4, 5, 6, 7, 8, 9}); err != nil {
		return err
	}
```

And batch insert with batch.Send(), which I believe is sync insert
```go
	batch, err := d.conn.PrepareBatch(context.Background(), "INSERT INTO example")
	if err != nil {
		return err
	}
	for i := 0; i < 1_000; i++ {
		err := batch.AppendStruct(&row{
			Col1:       uint64(i),
			Col2:       "Golang SQL database driver",
			Col3:       []uint8{1, 2, 3, 4, 5, 6, 7, 8, 9},
			Col4:       time.Now(),
			ColIgnored: "this will be ignored",
		})
		if err != nil {
			return err
		}
	}
	return batch.Send()
```

Is there a way to insert multiple rows async ?

**Comment by jkaflik** [comment]
> Is there a way to insert multiple rows async ?

@0x366 ClickHouse async insert is a query setting. Driver `AsyncInsert` is the usual query execution with an additional `async_insert` set for you. See: https://clickhouse.com/docs/en/cloud/bestpractices/asynchronous-inserts

You can execute async insert with `PrepareBatch` with additional setting set. Example of context with settings: https://github.com/ClickHouse/clickhouse-go/blob/a9eb5130847c0fdbbd845a388c3eeb522b68f56b/examples/clickhouse_api/context.go#L56-L58

This should be updated in documentation. We will follow up.
